### PR TITLE
fix: settle stale automation runs

### PR DESCRIPTION
**Pull request by carlosflorencio**
> [!TIP]
> **PR walkthrough:** [Open the visual walkthrough](https://walkthrough.kandev.ai/pr/3096/7a1b6acb7290.html)

Automation runs could remain open after a restart when their bound turn had already completed, blocking future firings. This change terminalizes genuinely open stale runs and treats known gone-execution liveness errors as settled runs during reconciliation.

## Validation

- `env -u KANDEV_INTERNAL_CONFIG_FILE -u KANDEV_INTERNAL_CONFIG_HOME_FILE -u KANDEV_HOME_DIR make test` (from `apps/backend`)
- `make lint` (from `apps/backend`)
- `golangci-lint run ./... --new-from-rev=c2e2b3a430d0bd865a3d0bc159e9d8543bc1b02d` (from `apps/backend`)
- `gofmt -l` on all changed Go files
- `python3 scripts/lint-spec-files.py --all`
- Focused table-driven tests for stale stopping, reconciliation, and gone-execution normalization

## Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have manually tested my changes and they work as expected.
- [ ] My changes have tests that cover the new functionality and edge cases.
- [ ] If my change touches UI files (`apps/web/`), I have added or updated Playwright e2e tests in `apps/web/e2e/` and verified them with `make test-e2e`.
- [ ] I checked whether this affects public docs in `docs/public/**` and updated them or noted why no docs change is needed.

<a href="https://cubic.dev/pr/kdlbs/kandev/pull/3096?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: apps/backend/internal/automation/service_run_terminal_test.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+func TestReconcileOpenRuns(t *testing.T) {
+	transientErr := errors.New("database temporarily unavailable")
+	tests := []struct {
+		name        string
+		bound       bool
+		live        bool
+		livenessErr error
+		wantStatus  RunStatus
+		wantMessage string
+	}{
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/backend/internal/automation/service_run_terminal_test.go:131]: Addressed in `7a1b6acb72`: removed the duplicate reconciliation case whose stub returned nil/false and kept the real normalization coverage in `TestAutomationRunLiveNormalizesGoneExecutionErrors`. The remaining service case is now named `not live settles run`.

File: apps/backend/internal/automation/service.go [hunk 2]
```diff
@@ -916,13 +910,10 @@ func (s *Service) StopRun(ctx context.Context, automationID, runID string) (*Aut
 		if s.runStopper == nil {
 			return nil, errors.New("automation run stopper is not configured")
 		}
-		stopped, stopErr := s.runStopper.StopAutomationRun(ctx, run.TaskID, run.SessionID, run.TurnID)
+		_, stopErr := s.runStopper.StopAutomationRun(ctx, run.TaskID, run.SessionID, run.TurnID)
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/backend/internal/automation/service.go:913]: Addressed in `7a1b6acb72`: `StopRun` now reloads the run when terminalization races with normal completion. If the row is already terminal, it returns that state as an idempotent success; errors remain surfaced when the row is still open or unreadable. Added the `completion settles concurrently` regression.

File: apps/backend/internal/orchestrator/event_handlers_automation.go [hunk 3]
```diff
@@ -99,6 +110,12 @@ func (s *Service) AutomationRunLive(ctx context.Context, taskID, sessionID, turn
 	return s.hasPendingCoordinatorPermissions(ctx, sessionID)
 }
 
+func automationRunExecutionGone(err error) bool {
+	return runtimeapi.IsNotFound(err) ||
+		errors.Is(err, executor.ErrExecutionNotFound) ||
+		errors.Is(err, sql.ErrNoRows)
```
[the remark this answers is not in the document]
Comment (carlosflorencio) [reply apps/backend/internal/orchestrator/event_handlers_automation.go]: Addressed in `7a1b6acb72`: `AutomationRunLive` now classifies wrapped task-repository and task-session not-found sentinels as gone bindings alongside runtime, executor, lifecycle, and SQL sentinels. Added real SQLite coverage for a missing task and a deleted session. Reconciliation then settles the run.
